sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/util/ParseMode.scala

```diff
@@ -17,25 +17,35 @@
 
 package org.apache.spark.sql.catalyst.util
 
-object ParseModes {
-  val PERMISSIVE_MODE = "PERMISSIVE"
-  val DROP_MALFORMED_MODE = "DROPMALFORMED"
-  val FAIL_FAST_MODE = "FAILFAST"
+import org.apache.spark.internal.Logging
 
-  val DEFAULT = PERMISSIVE_MODE
+object ParseMode extends Enumeration with Logging {
```

Comment:
seems people usually use `sealed trait` and `case object` to implement enum in scala, see http://stackoverflow.com/questions/1898932/case-objects-vs-enumerations-in-scala